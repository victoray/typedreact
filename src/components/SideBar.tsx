import React from "react";
import {Link} from "react-router-dom";

const SideBar: React.FC = () => {
    return (
        <div className="ui left fixed vertical menu">
            <div className="item">
                <img className="ui mini image" alt={""} src={"https://semantic-ui.com/images/logo.png"}/>
            </div>
            <Link className="item" to={"/"}>DashBoard <i className="chart line icon"/></Link>
            <Link className="item" to={"/"}>StarWars <i className="star icon"/></Link>
            <Link className="item" to={"/"}>Avengers <i className="nintendo switch icon"/></Link>
            <Link className="item" to={"/"}>Settings <i className="cogs icon"/></Link>
        </div>);
};

export default SideBar;
