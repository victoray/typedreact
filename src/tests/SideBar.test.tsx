import React from "react";
import renderer from 'react-test-renderer';
import SideBar from "../components/SideBar";
import {BrowserRouter} from "react-router-dom";


it('should not be visible', function () {
    const tree = renderer
        .create(<BrowserRouter><SideBar visible={false}/></BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('should be visible', function () {
    const tree = renderer
        .create(<BrowserRouter><SideBar visible={true}/></BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});