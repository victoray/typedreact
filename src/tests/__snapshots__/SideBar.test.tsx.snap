// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be visible 1`] = `
<div
  className="ui visible sidebar left fixed vertical menu"
>
  <div
    className="item"
  >
    <img
      alt=""
      className="ui mini image"
      src="https://semantic-ui.com/images/logo.png"
    />
  </div>
  <a
    className="item"
    href="/dashboard"
    onClick={[Function]}
  >
    DashBoard 
    <i
      className="chart line icon"
    />
  </a>
  <a
    className="item"
    href="/starwars"
    onClick={[Function]}
  >
    StarWars 
    <i
      className="star icon"
    />
  </a>
  <a
    className="item"
    href="/avengers"
    onClick={[Function]}
  >
    Avengers 
    <i
      className="nintendo switch icon"
    />
  </a>
  <a
    className="item"
    href="/settings"
    onClick={[Function]}
  >
    Settings 
    <i
      className="cogs icon"
    />
  </a>
</div>
`;

exports[`should not be visible 1`] = `
<div
  className="ui  sidebar left fixed vertical menu"
>
  <div
    className="item"
  >
    <img
      alt=""
      className="ui mini image"
      src="https://semantic-ui.com/images/logo.png"
    />
  </div>
  <a
    className="item"
    href="/dashboard"
    onClick={[Function]}
  >
    DashBoard 
    <i
      className="chart line icon"
    />
  </a>
  <a
    className="item"
    href="/starwars"
    onClick={[Function]}
  >
    StarWars 
    <i
      className="star icon"
    />
  </a>
  <a
    className="item"
    href="/avengers"
    onClick={[Function]}
  >
    Avengers 
    <i
      className="nintendo switch icon"
    />
  </a>
  <a
    className="item"
    href="/settings"
    onClick={[Function]}
  >
    Settings 
    <i
      className="cogs icon"
    />
  </a>
</div>
`;
